lib/rouge/lexers/lutin.rb

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*- #
+#
+# adapted from lustre.rf (adapted from ocaml.rb), hence some ocaml-ism migth remains
+module Rouge
+  module Lexers
+    class Lutin < RegexLexer
+      title "Lutin"
+      desc 'The Lutin programming language (Verimag)'
+      tag 'lutin'
+      filenames '*.lut'
+      mimetypes 'text/x-lutin'
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          let in node extern system returns weak strong assert raise try catch
+          trap do exist erun run type ref exception include false true 
+        )
+      end
+
+      def self.word_operators
+        @word_operators ||= Set.new %w(
+           div and xor mod or not nor if then else pre) 
+      end
+
+      def self.primitives
+        @primitives ||= Set.new %w(int real bool trace loop fby)
+      end
+
+      operator = %r([;,_!$%&*+./:<=>?@^|~#-]+)
+      id = /[a-z_][\w']*/i
+
+      state :root do
+        rule /\s+/m, Text
+        rule /false|true|[(][)]|\[\]/, Name::Builtin::Pseudo
+        rule %r(\-\-.*?$), Comment::Single
+        rule %r(/\*.*?\*/)m, Comment::Multiline
+        rule %r(\(\*.*?\*\))m, Comment::Multiline
+        rule id do |m|
+          match = m[0]
+          if self.class.keywords.include? match
+            token Keyword
+          elsif self.class.word_operators.include? match
+            token Operator::Word
+          elsif self.class.primitives.include? match
+            token Keyword::Type
+          else
+            token Name
+          end
+        end
+
+        rule /[(){}\[\];]+/, Punctuation
+        rule operator, Operator
+
+        rule /-?\d[\d_]*(.[\d_]*)?(e[+-]?\d[\d_]*)/i, Num::Float
+        rule /0x\h[\h_]*/i, Num::Hex
+        rule /0o[0-7][0-7_]*/i, Num::Oct
+        rule /0b[01][01_]*/i, Num::Bin
+        rule /\d[\d_]*/, Num::Integer
+
+        rule /'(?:(\\[\\"'ntbr ])|(\\[0-9]{3})|(\\x\h{2}))'/, Str::Char
+        rule /'[.]'/, Str::Char
+        rule /'/, Keyword
+        rule /"/, Str::Double, :string
+        rule /[~?]#{id}/, Name::Variable
+      end
+
+      state :string do
+        rule /[^\\"]+/, Str::Double
+        mixin :escape_sequence
+        rule /\\\n/, Str::Double
+        rule /"/, Str::Double, :pop!
+      end
+
+      state :escape_sequence do
+        rule /\\[\\"'ntbr]/, Str::Escape
+        rule /\\\d{3}/, Str::Escape
+        rule /\\x\h{2}/, Str::Escape
+      end
+
+      state :dotted do
```

Comment:
This state isn't pushed onto the stack or mixed in. Is that an oversight or is it from an older version of the code and can now be deleted?

Reply:
I'll adress this in the lutin dedicated PR.

Reply:
cf PR#58

Reply:
I'm a bit confused—this state is still in the lexer and is still never used. it should be deleted, right?

Reply:
Why do you say that? I've removed it.
https://github.com/rouge-ruby/rouge/pull/1307/commits/7c62be91edf84b8cf4f5a710495d35dbdcdfdd56

Reply:
I'm talking about the `:dotted` state in [the Lustre lexer](https://github.com/jahierwan/rouge/blob/3ee37a619c53810217e043a965820610ba505383/lib/rouge/lexers/lustre.rb#L78). It's still there and isn't ever used.

Reply:
oh, ok. I did not understand in the first place because this is a comment of lutin.rb. I will remove it from lustre.rb too, thanks.